openshift/s2i/bin/assemble

```diff
@@ -0,0 +1,91 @@
+#!/bin/bash
+
+function is_django_installed() {
+  python -c "import django" &>/dev/null
+}
+
+function should_collectstatic() {
+  is_django_installed && [[ -z "$DISABLE_COLLECTSTATIC" ]]
+}
+
+function virtualenv_bin() {
+  if head "/etc/redhat-release" | grep -q "^Fedora"; then
+    virtualenv-${PYTHON_VERSION} $1
+  else
+    virtualenv $1
+  fi
+}
+
+# Install pipenv to the separate virtualenv to isolate it
+# from system Python packages and packages in the main
+# virtualenv. Executable is simlinked into ~/.local/bin
+# to be accessible. This approach is inspired by pipsi
+# (pip script installer).
+function install_pipenv() {
+  echo "---> Installing pipenv packaging tool ..."
+  VENV_DIR=$HOME/.local/venvs/pipenv
+  virtualenv_bin "$VENV_DIR"
+  $VENV_DIR/bin/pip --isolated install -U pip==18.0
+  $VENV_DIR/bin/pip --isolated install -U pipenv==2018.6.25
+  mkdir -p $HOME/.local/bin
+  ln -s $VENV_DIR/bin/pipenv $HOME/.local/bin/pipenv
+}
+
+set -e
+
+shopt -s dotglob
+echo "---> Installing application source ..."
+mv /tmp/src/* ./
+
+if [[ ! -z "$UPGRADE_PIP_TO_LATEST" ]]; then
```

Comment:
Add `UPGRADE_PIP_TO_LATEST` and `ENABLE_PIPENV` to an env file or the template?

Reply:
`ENABLE_PIPENV` is already in the buildconfigs, but I'll add `UPGRADE_PIP_TO_LATEST`.